libraries/node-core-library/src/Executable.ts

```diff
@@ -113,11 +114,221 @@ interface IExecutableContext {
   windowsExecutableExtensions: string[];
 }
 
-interface ICommandLineFixup {
+interface ICommandLineOptions {
   path: string;
   args: string[];
 }
 
+/**
+ * Process information sourced from the system. This process info is sourced differently depending
+ * on the operating system:
+ * - On Windows, this uses the `wmic.exe` utility.
+ * - On Unix, this uses the `ps` utility.
+ *
+ * @public
+ */
+export interface IProcessInfo {
+  /**
+   * The name of the process.
+   *
+   * @remarks On Windows, the process name will be empty if the process is a kernel process.
+   * On Unix, the process name will be empty if the process is the root process.
+   */
+  processName: string;
+  /**
+   * The process ID.
+   */
+  processId: number;
+  /**
+   * The parent process info.
+   *
+   * @remarks On Windows, the parent process info will be undefined if the process is a kernel process.
+   * On Unix, the parent process info will be undefined if the process is the root process.
+   */
+  parentProcessInfo?: IProcessInfo;
+
+  /**
+   * The child process infos.
+   */
+  childProcessInfos: IProcessInfo[];
+}
+
+// Match the newline character(s) at the end of a line of text in order to split on it. On Windows,
+// it is possible for multiple \r characters may precede the \n character, so we need to match all
+// of them.
+const NEWLINE_REGEX: RegExp = /\r*\n/;
+
+async function* readLinesFromStreamAsync(stream: NodeJS.ReadableStream): AsyncGenerator<string> {
```

Comment:
```suggestion
async function* readLinesFromStreamAsync(stream: AsyncIterable<string>): AsyncGenerator<string> {
```
Since you're only using the `for await` code path.